src/chalk_common.nim

```diff
@@ -478,14 +478,17 @@ var
   dockerInvocation*:      DockerInvocation
 
 template dumpExOnDebug*() =
-  if getChalkScope() != nil and get[bool](getChalkScope(), "chalk_debug"):
-    let
-      msg = "" # "Handling exception (msg = " & getCurrentExceptionMsg() & ")\n"
-      tb  = "Traceback (most recent call last)\n" &
-             getCurrentException().getStackTrace()
-      ii  = default(InstInfo)
-
-    publish("debug", formatCompilerError(msg, nil, tb, ii))
+  when not defined(release):
+    let stack = getCurrentException().getStackTrace()
+    if getChalkScope() != nil and get[bool](getChalkScope(), "chalk_debug"):
+      let
+        msg = "" # "Handling exception (msg = " & getCurrentExceptionMsg() & ")\n"
+        tb  = "Traceback (most recent call last)\n" & stack
+        ii  = default(InstInfo)
+        fmt = formatCompilerError(msg, nil, tb, ii)
+      publish("debug", fmt)
+    else:
+      trace(stack)
```

Comment:
without `--debug` it shows up as:

<img width="1792" alt="image" src="https://github.com/user-attachments/assets/76b28f93-26ef-4c94-8917-a17b8f74492a">


with `--debug` normal formatting applies:

<img width="2069" alt="image" src="https://github.com/user-attachments/assets/890afd59-8f8c-4205-92dd-6e31454f6d2e">